### ISSUE TITLE
Over-stripping of ANSI formatting.

### ISSUE THREAD

**Issue by kdeldycke** [mahmoud/boltons#257]
I just stumble upon a case of `strip_ansi()` method being overzealous in stripping formatting escape sequences:

```python
>>> from boltons.strutils import strip_ansi
>>> table
('╒════════════════╤═════════╤═══════════╤══════════════════╕\n'
 '│ Package name   │ ID      │ Manager   │ Latest version   │\n'
 '╞════════════════╪═════════╪═══════════╪══════════════════╡\n'
 '│ \x1b[1msed\x1b[0m            │ \x1b[1msed\x1b[0m     │ pip3      │ '
 '0.3.1            │\n'
 '├────────────────┼─────────┼───────────┼──────────────────┤\n'
 '│ SED-cli        │ SED-cli │ pip3      │ 0.0.1            │\n'
 '╘════════════════╧═════════╧═══════════╧══════════════════╛')
>>> print(table)
╒════════════════╤═════════╤═══════════╤══════════════════╕
│ Package name   │ ID      │ Manager   │ Latest version   │
╞════════════════╪═════════╪═══════════╪══════════════════╡
│ sed            │ sed     │ pip3      │ 0.3.1            │
├────────────────┼─────────┼───────────┼──────────────────┤
│ SED-cli        │ SED-cli │ pip3      │ 0.0.1            │
╘════════════════╧═════════╧═══════════╧══════════════════╛
>>> strip_ansi(table)
('╒════════════════╤═════════╤═══════════╤══════════════════╕\n'
 '│ Package name   │ ID      │ Manager   │ Latest version   │\n'
 '╞════════════════╪═════════╪═══════════╪══════════════════╡\n'
 '│ sed            │ sed')
>>> print(strip_ansi(table))
╒════════════════╤═════════╤═══════════╤══════════════════╕
│ Package name   │ ID      │ Manager   │ Latest version   │
╞════════════════╪═════════╪═══════════╪══════════════════╡
│ sed            │ sed
>>> 
```

This is rendered as such in a console. You can see how some substrings are highlighted in the table:

<img width="565" alt="Screen Shot 2020-05-14 at 15 50 40" src="https://user-images.githubusercontent.com/159718/81942730-cce2ce00-95fa-11ea-9114-02c68206d69a.png">

Here I expect the result of `strip_ansi(table)` to returns:
```
('╒════════════════╤═════════╤═══════════╤══════════════════╕\n'
 '│ Package name   │ ID      │ Manager   │ Latest version   │\n'
 '╞════════════════╪═════════╪═══════════╪══════════════════╡\n'
 '│ sed            │ sed     │ pip3      │ 0.3.1            │\n'
 '├────────────────┼─────────┼───────────┼──────────────────┤\n'
 '│ SED-cli        │ SED-cli │ pip3      │ 0.0.1            │\n'
 '╘════════════════╧═════════╧═══════════╧══════════════════╛')
```

I can try fixing that with a PR but I need to have a confirmation my proposed output is the intended one.

**Comment by mahmoud** [comment]
Hmm, yes, I believe it, and would be more than happy to review a PR for this.

**Comment by mahmoud** [comment]
Released last night in 20.2.0! :tada:
